src/message/messageActionSheet.js

```diff
@@ -252,10 +261,16 @@ export const constructTopicActionButtons = ({
     buttons.push(muteTopic);
   }
   const sub = subscriptions.find(x => x.name === stream);
-  if (sub && !sub.in_home_view) {
-    buttons.push(unmuteStream);
-  } else {
-    buttons.push(muteStream);
+  if (sub) {
+    const unreadCount = unreadStreams.get(sub.stream_id)?.get(topic)?.size;
+    if (unreadCount !== undefined && unreadCount > 0) {
```

Comment:
Either `if (unreadCount) {` or `if (unreadCount && unreadCount > 0) {` should work here. I'd go with the second, even though it's redundant (well, unless we have a negative number of unreads, in which case we have a second problem...), just since it reads more clearly to me.

Reply:
On doing this, flow gives me 2 error.
> Avoid using \`&&\` to check the value of  number [1]. Consider handling falsy values (0 and NaN) by using a conditional to choose an explicit default instead.

> Sketchy null check on  number [1] which is potentially 0. Perhaps you meant to check for  null or undefined [2]?

Your condition checks if `unreadCount` gives a truthy value which is a bad idea for number data type since 0 results in a falsy value (hence we have this rule set in our `.flowconfig`).
I understand that it won't affect this condition but this is a special case, hence, to avoid the error essentially, I am checking for `undefined` explicitly.

Reply:
Ah, I see. I'd personally go with `if (!!unreadCount && unreadCount > 0) {`, but it doesn't really matter, since I don't think that `unreadCount` can be `null`. Seems fine as-is.